src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -287,10 +287,13 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 if (isRMW)
                 {
-                    assert(targetReg != op2Reg);
-                    assert(targetReg != op3Reg);
+                    if (targetReg != op1Reg)
```

Comment:
Not sure how this solves the problem. We `emitIns_Mov` already skips the move (see `/* canSkip */ true`) if `targetReg == op1Reg`.

Reply:
That's correct (that the `mov` is already elided). It solves the problem by putting the asserts inside the `if`.

Reply:
Ah, I missed the part where you said: "So, put the asserts under a check that the mov is required.".  But regardless, I think we should fix the problem of updating `delayRegFree` properly based on `lastUse` information.

Reply:
> But regardless, I think we should fix the problem of updating `delayRegFree` properly based on `lastUse` information.

I agree. But I'm not sure it's related to this fix: we still should be able to generate `mla` with all the same registers as arguments (however weird a case that would be).